src/field/field3d.cxx

```diff
@@ -229,14 +229,19 @@ const Field3D& Field3D::ynext(int dir) const {
   }
 }
 
-void Field3D::setLocation(CELL_LOC loc) {
-  if(loc == CELL_VSHIFT)
-    throw BoutException("Field3D: CELL_VSHIFT cell location only makes sense for vectors");
-  
-  if(loc == CELL_DEFAULT)
-    loc = CELL_CENTRE;
-  
-  location = loc;
+void Field3D::setLocation(CELL_LOC new_location) {
+  if (getMesh()->StaggerGrids) {
+    if (new_location == CELL_VSHIFT) {
+      throw BoutException(
+          "Field3D: CELL_VSHIFT cell location only makes sense for vectors");
+    }
+    if (new_location == CELL_DEFAULT) {
+      new_location = CELL_CENTRE;
+    }
+    location = new_location;
+  } else {
+    location = CELL_CENTRE;
```

Comment:
Should we check that we don't try to set a non-centre location in the non-staggered case and if we do then throw? This would possibly help us catch places where we try to set staggering but don't check that we want to stagger? (I don't think this is my idea, I'm sure it's been mentioned previously)

Reply:
I agree at least a warning here would be appropriate, along with modifying all current calling sites to check if we're using staggered grids.